### PR TITLE
Add post surveys endpoint

### PR DESCRIPTION
**Pull request by plapicola**
Adds initial endpoint for creation of surveys. Endpoint returns a 201 if successful or a 422 if the request is unable to be processed. If invalid users are provided for groups, survey creation will still succeed. Testing is present for happy path.

@aprildagonese How do we want to handle the concept of a template? Do we need a new field on the survey post to indicate it is a template?

Also, the timestamp is still work in progress. I'll get that added and submit in a separate PR.

### CODE REVIEW

File: lib/feedback_api_web/facades/survey_create_facade.ex [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    Enum.map(groups, fn group -> create_group(survey, group) end)
+  end
+
+  defp create_questions(survey, questions) do
+    Enum.map(questions, fn question -> create_question(survey, question) end)
+  end
+
+  defp create_group(survey, group) do
+    new_group =
+      Group.changeset(%Group{}, group) |> Repo.insert!() |> Repo.preload([:survey, :users])
```
Comment (aprildagonese) [inline lib/feedback_api_web/facades/survey_create_facade.ex]: What is Repo.preload doing here?

File: lib/feedback_api_web/facades/survey_create_facade.ex [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    users = Repo.all(from u in User, where: u.id in ^group["members_ids"])
```
Comment (aprildagonese) [inline lib/feedback_api_web/facades/survey_create_facade.ex]: Could you spell out line 29 for me?

File: lib/feedback_api_web/facades/survey_create_facade.ex [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    group_associations_changeset =
+      Ecto.Changeset.change(new_group)
+      |> Ecto.Changeset.put_assoc(:survey, survey)
```
Comment (aprildagonese) [inline lib/feedback_api_web/facades/survey_create_facade.ex]: Just to make sure--is this just declaring the associations explicitly? And you have to do that in a separate changeset?

File: priv/repo/migrations/20190519175341_add-name-to-groups.exs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+defmodule :"Elixir.FeedbackApi.Repo.Migrations.Add-name-to-groups" do
```
Comment (aprildagonese) [inline priv/repo/migrations/20190519175341_add-name-to-groups.exs]: Looks like it doesn't love dashes in migration names.

File: lib/feedback_api_web/controllers/surveys_controller.ex [hunk 5]
```diff
@@ -1,6 +1,7 @@
 defmodule FeedbackApiWeb.SurveysController do
   use FeedbackApiWeb, :controller
   alias FeedbackApi.{Survey, Repo}
+  alias FeedbackApiWeb.SurveyCreateFacade
 
   def index(conn, _params) do
     surveys = Survey |> Repo.all()
```
[the remark this answers is not in the document]
Comment (plapicola) [reply lib/feedback_api_web/controllers/surveys_controller.ex]: I think the latter is allowed, but this was just the way I saw it in the docs happy to change it if the latter makes more sense.

File: lib/feedback_api_web/facades/survey_create_facade.ex (hunk 1 above)
Comment (plapicola) [reply lib/feedback_api_web/facades/survey_create_facade.ex]: It is loading the associated records, which is a requirement to have it build the relationships on the lines below. Otherwise the object will come back without the fields for the associations.

File: lib/feedback_api_web/facades/survey_create_facade.ex (hunk 2 above)
Comment (plapicola) [reply lib/feedback_api_web/facades/survey_create_facade.ex]: It's finding all records (u) in the Users table where the primary key is in the array of group members from the request.

File: lib/feedback_api_web/facades/survey_create_facade.ex (hunk 3 above)
Comment (plapicola) [reply lib/feedback_api_web/facades/survey_create_facade.ex]: As far as I know, yes. You cannot directly assign foreign key values, and must include an association changeset, which is then committed.

File: priv/repo/migrations/20190519175341_add-name-to-groups.exs (hunk 4 above)
Comment (plapicola) [reply priv/repo/migrations/20190519175341_add-name-to-groups.exs]: Yeah, I'll be more cognizant of that going forward, sorry.
